grpc_publisher_service.py

```diff
@@ -513,6 +514,83 @@ def get_platform_info(platform_letter):
                     send_to_sentry=True,
                 )
             return platform_info, None
+        
+        def handle_test_client(
+            msisdn,
+            payload,
+            context,
+            response,
+        ):
+          
+            try:
+                sms_sent_time_epoch = payload.get("time")
+                test_id = payload.get("id")
+
+                try:
+                    test_id = int(test_id)
+                    logger.debug(f"Converted test_id to integer: {test_id}")
+                except ValueError:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        f"Invalid test_id: {test_id}. Must be an integer.",
+                        grpc.StatusCode.INVALID_ARGUMENT,
+                        error_prefix="Invalid Payload",
+                        send_to_sentry=True,
+                    )
+
+                if not sms_sent_time_epoch or not test_id:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        "Invalid payload: Missing required fields (time or id).",
+                        grpc.StatusCode.INVALID_ARGUMENT,
+                        error_prefix="Invalid Payload",
+                        send_to_sentry=True,
+                    )
```

Comment:
No need for this check here. 
It should rather be done when extracting the content parts